src/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -1421,8 +1421,7 @@ public void SymmetricExceptWith(IEnumerable<T> other)
             }
             else
             {
-                //need perf improvement on this
-                T[] elements = (new List<T>(other)).ToArray();
+                T[] elements = EnumerableHelpers.ToArray(other);
```

Comment:
I wonder if it'd be worth using the other `EnumerableHelpers.ToArray` overload to avoid the (potential) extra array allocation to downsize the array, e.g. something like:

```c#
int length;
T[] elements = EnumerableHelpers.ToArray(other, out length);
Array.Sort(elements, 0, length, Comparer);
SymmetricExceptWithSameEC(elements, length);
```

...with a modified `SymmetricExceptWithSameEC(T[] other, int length)` that takes the array length.